app/src/main/java/swati4star/createpdf/util/InvertPdf.java

```diff
@@ -0,0 +1,118 @@
+package swati4star.createpdf.util;
+
+import android.graphics.Bitmap;
+import android.os.AsyncTask;
+import android.os.ParcelFileDescriptor;
+import com.itextpdf.text.Rectangle;
+import com.itextpdf.text.pdf.GrayColor;
+import com.itextpdf.text.pdf.PdfContentByte;
+import com.itextpdf.text.pdf.PdfGState;
+import com.itextpdf.text.pdf.PdfReader;
+import com.itextpdf.text.pdf.PdfStamper;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+import swati4star.createpdf.interfaces.OnPDFCreatedInterface;
+
+import static android.os.ParcelFileDescriptor.MODE_READ_ONLY;
+
+public class InvertPdf extends AsyncTask<Void, Void, Void> {
+    private String mPath;
+    private OnPDFCreatedInterface mOnPDFCreatedInterface;
+    private ArrayList<Bitmap> mBitmaps;
+    private StringBuilder mSequence;
+    private Boolean mIsNewPDFCreated;
+
+    public InvertPdf(String mPath, OnPDFCreatedInterface onPDFCreatedInterface) {
+        this.mPath = mPath;
+        mSequence = new StringBuilder();
+        mBitmaps = new ArrayList<>();
+        this.mOnPDFCreatedInterface = onPDFCreatedInterface;
+    }
+
+    @Override
+    protected void onPreExecute() {
+        super.onPreExecute();
+        mOnPDFCreatedInterface.onPDFCreationStarted();
+        mIsNewPDFCreated = false;
+    }
+
+
+    @Override
+    protected Void doInBackground(Void... voids) {
+        ParcelFileDescriptor fileDescriptor = null;
+        try {
+            if (mPath != null)
+                // resolve pdf file path based on relative path
+                fileDescriptor = ParcelFileDescriptor.open(new File(mPath), MODE_READ_ONLY);
+
+
+            if (fileDescriptor != null) {
+                String outputPath = mPath.replace(".pdf", "_inverted" + ".pdf");
+                if (createPDF(mPath, outputPath)) {
+                    mPath = outputPath;
+                    mIsNewPDFCreated = true;
+                }
+
+            }
+        } catch (IOException | SecurityException e) {
+            e.printStackTrace();
+            mIsNewPDFCreated = false;
+        }
+
+        return null;
+    }
+
+    void invert(PdfStamper stamper) {
+        for (int i = stamper.getReader().getNumberOfPages(); i > 0; i--) {
+            invertPage(stamper, i);
+        }
+    }
+
+    void invertPage(PdfStamper stamper, int page) {
```

Comment:
add javadoc style comments

Reply:
Added a little bit of documentation and also changed a little bit of structure(putting onPost at bottom) for easier reading.
Let me know if there's something else.
:)